### PR TITLE
Jcs/database tests

### PR DESCRIPTION
**Pull request by jSylvestre**

### CODE REVIEW

File: Test/TestsDatabase/OrderDatabaseTests.cs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+using Test.Helpers;
+using Xunit;
+
+namespace Test.TestsDatabase
+{
+    [Trait("Category", "DatabaseTests")]
+    public class OrderDatabaseTests
+    {
+        [Fact]
+        public void test1()
```
Comment (srkirkland) [inline Test/TestsDatabase/OrderDatabaseTests.cs]: @jSylvestre is there such a thing as beforeTest/afterTest w/ xUnit?  Seems like opening/closing the connection would be a good thing to do there and would save a lot of boilerplate

File: Test/TestsDatabase/OrderDatabaseTests.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            try
+            {
+                var options = new DbContextOptionsBuilder<ApplicationDbContext>()
+                    .UseSqlite(connection)
+                    .Options;
+
+                using (var context = new ApplicationDbContext(options))
+                {
+                    context.Database.EnsureCreated();
+                }
```
Comment (srkirkland) [inline Test/TestsDatabase/OrderDatabaseTests.cs]: @jSylvestre what is the point of using multiple contexts?

Comment (jSylvestre) [inline Test/TestsDatabase/OrderDatabaseTests.cs]: @srkirkland It was in the test example:
https://docs.microsoft.com/en-us/ef/core/miscellaneous/testing/sqlite

File: Test/TestsDatabase/OrderDatabaseTests.cs (hunk 1 above)
Comment (jSylvestre) [inline Test/TestsDatabase/OrderDatabaseTests.cs]: @srkirkland Yes, but:
Note 2: The xUnit.net team feels that per-test setup and teardown creates difficult-to-follow and debug testing code, often causing unnecessary code to run before every single test is run. For more information, see http://jamesnewkirk.typepad.com/posts/2007/09/why-you-should-.html.

Comment (srkirkland) [inline Test/TestsDatabase/OrderDatabaseTests.cs]: @jSylvestre ok, so maybe just a helper method that creates the db and opens a context would work better

File: Test/TestsDatabase/OrderDatabaseTests.cs (hunk 2 above)
Comment (srkirkland) [inline Test/TestsDatabase/OrderDatabaseTests.cs]: @jSylvestre Try it without and see if it makes any difference.  Seems like extra work to me

File: Test/TestsDatabase/OrderDatabaseTests.cs (hunk 1 above)
Comment (srkirkland) [inline Test/TestsDatabase/OrderDatabaseTests.cs]: @jSylvestre something like a class that implements iDisposible and opens up a new connection and wraps everything in a try/catch/finally.

**Pull request merged**